### PR TITLE
Feature/config

### PR DESCRIPTION
**Pull request by spellgen**
See what you think about style and naming

### CODE REVIEW

File: configurator_test.go [hunk 1]
```diff
@@ -62,14 +104,22 @@ func (c *config) Update(_ uint64, data interface{}) {
...
 	default:
 		c.t.Log("Typecast failed")
 		c.t.Fail()
 	}
 }
 
+func (c *config) copy() *config {
+	c.RLock()
+	defer c.RUnlock()
+	cc := *c
```
Comment (aaronhurt) [inline configurator_test.go]: This is failing govet...

```

Comment (aaronhurt) [inline configurator_test.go]: This is failing govet in the travis build: https://travis-ci.org/myENA/consultant/jobs/236450500

```
configurator_test.go:119: assignment copies lock value to cc: consultant_test.config
```

**Comment by spellgen** [comment]
doing this the go way instead - chan chan chan all the way

**Comment by dcarbone** [comment]
lgtm :)

**Pull request merged**